pulp_rpm/tests/functional/api/test_domains.py

```diff
@@ -0,0 +1,533 @@
+            )
+            repos = rpm_repository_api.list(pulp_domain=domain.name)
+            assert repos.count == 1
+            assert repos.results[0].pulp_href == repo.pulp_href
+            # Try to create a repository in default domain
+            with pytest.raises(ApiException) as e:
+                rpm_repository_api.create({"name": str(uuid.uuid4())})
+            assert e.value.status == 403
+
+        with user_a:
+            repos = rpm_repository_api.list(pulp_domain=domain.name)
+            assert repos.count == 2
+            # Try to read repos in the default domain
+            repos = rpm_repository_api.list()
+            assert repos.count == 0
+            # Try to create a repo
+            with pytest.raises(ApiException) as e:
+                rpm_repository_api.create({"name": str(uuid.uuid4())}, pulp_domain=domain.name)
+            assert e.value.status == 403
+
+    finally:
+        cleanup_domains([domain], cleanup_repositories=True)
+
+
+@pytest.mark.parallel
+def test_advisory_upload_json(
+    cleanup_domains,
+    setup_domain,
+    upload_advisory_factory,
+):
+    """Test upload same advisory from JSON file into different Domains."""
+    domain1 = None
+    domain2 = None
+    try:
+        domain1, _, repo1, _ = setup_domain(sync=False)
+        domain2, _, repo2, _ = setup_domain(sync=False)
+
+        advisory1, _, an_id = upload_advisory_factory(
+            pulp_domain=domain1.name, repository=repo1, set_id=True
+        )
+        assert advisory1.id == an_id
+
+        advisory2, _, an_id = upload_advisory_factory(
+            pulp_domain=domain2.name, repository=repo2, set_id=True
+        )
+        assert advisory2.id == an_id
+
+        assert advisory1.pulp_href != advisory2.pulp_href
+    finally:
+        cleanup_domains([domain1, domain2], cleanup_repositories=True)
+
+
+@pytest.mark.parallel
+def test_cross_domain_copy_all(
+    monitor_task,
+    rpm_copy_api,
+    cleanup_domains,
+    setup_domain,
+):
+    """Test attempting to copy between different Domains."""
+    domain1 = None
+    domain2 = None
+    try:
+        domain1, remote1, src1, dest1 = setup_domain()
+        domain2, remote2, src2, dest2 = setup_domain()
+        # Success, everything in domain1
+        data = Copy(
+            config=[
+                {"source_repo_version": src1.latest_version_href, "dest_repo": dest1.pulp_href}
+            ],
+            dependency_solving=False,
+        )
+        monitor_task(rpm_copy_api.copy_content(data, pulp_domain=domain1.name).task)
+
+        # Failure, call and src domain1, dest domain2
+        with pytest.raises(ApiException):
+            data = Copy(
+                config=[
+                    {"source_repo_version": src1.latest_version_href, "dest_repo": dest2.pulp_href}
+                ],
+                dependency_solving=False,
+            )
+            monitor_task(rpm_copy_api.copy_content(data, pulp_domain=domain1.name).task)
```

Comment:
Don't think you need to call `monitor_task` here. The api call should error out before we get there.